test/functionality/slicing/reconstruct/function-tests.ts

```diff
@@ -0,0 +1,73 @@
+import { assert } from 'chai'
+import { plain, Code, merge, prettyPrintCodeToString } from '../../../../src/reconstruct/helper'
+
+describe('Functions_Reconstruct', () => {
+	describe('plain', () => {
+		function positive(input: string, line: number, column: number, expected: Code) {
+			it(`${input} for ${line}:${column}`, () => {
+				const result:Code = plain(input, {line,column})
+				assert.deepStrictEqual(result,expected)
+			})
+		}
+
+		for(const testCase of [
+			{input: 'Hello', line: 1, column: 1, expected: [{linePart: [{part: 'Hello', loc: {line: 1, column: 1}}],indent: 0}]},
+			{input: 'Hello World', line: 4, column: 3, expected: [{linePart: [{part: 'Hello World', loc: {line: 4, column: 3}}],indent: 0}]},
+			{input: 'Hello\nWorld', line: 1, column: 1, expected: [{linePart: [{part: 'Hello', loc: {line: 1, column: 1}},{part: 'World', loc: {line: 2, column: 1}}],indent: 0}]},
+			{input: 'Hello\nWorld', line: 3, column: 4, expected: [{linePart: [{part: 'Hello', loc: {line: 3, column: 4}},{part: 'World', loc: {line: 4, column: 4}}],indent: 0}]},
+			{input: 'Hello\nWorld\n24', line: 1, column: 1, expected: [{linePart: [{part: 'Hello', loc: {line: 1, column: 1}},{part: 'World', loc: {line: 2, column: 1}},{part: '24', loc: {line: 3, column: 1}}],indent: 0}]}
+		]) {
+			positive(testCase.input, testCase.line, testCase.column, testCase.expected)
+		}
+	})
+	describe.only('merge', () => {
+		function positive(snipbits: Code[],expected: Code) {
+			it(prettyPrintCodeToString(expected),() => {
+				const result:Code = merge(snipbits)
+				//console.log(JSON.stringify(result))
+				assert.deepStrictEqual(result, expected)
+			})
+		}
+		describe('single line merge', () => {
+			for(const testCase of [
+				{snipbit: [[{linePart: [{part: 'Hello World', loc: {line: 4, column: 3}}],indent: 0}]], expected: [{linePart: [{part: 'Hello World', loc: {line: 4, column: 3}}],indent: 0}]},
```

Comment:
woher kommt denn `snipbit`? :D

Reply:
Also ich mein als Bezeichner.

Reply:
Da der input für merge und der expected output einen unterschiedlichen Typ besitzen, habe ich mir für die Test Cases ein impiziertes Objekt erstellt das beides enthält (für den for-of-loop). snipbit kommt von meinem parameter snipbits in merge und könnte hier theoretisch auch input oder ähnliches genannt werden. :)

Reply:
snipbit enthält also nur meine eingabe für den testfall

Reply:
ja mir gings eher nur darum wie du auf den Namen gekommen bist :D

Reply:
weil merge ja code snipbits zusammen fügen soll😅

Reply:
meinst du snippets? oder verstehe ich den witz nicht :sweat_smile:

Reply:
schreibt man das so?😅

Reply:
Also wenn du ein Code-Schnipsel/einen Code-Teil meinst dann ja - mir ist snipbit zumindest noch _nie_ untergekommen :D

Reply:
dann ist das ein typo:)
aber ich glaub ich lass das vorerst mal so